jackson-databind/src/main/java/io/micronaut/jackson/modules/BeanIntrospectionModule.java

```diff
@@ -422,14 +422,18 @@ public Object getValue(Object pojo) throws IllegalArgumentException {
                             // we can't support XmlBeanPropertyWriter easily https://github.com/micronaut-projects/micronaut-core/issues/5907
                             !existing.getClass().getName().equals("com.fasterxml.jackson.dataformat.xml.ser.XmlBeanPropertyWriter")) { // NOSONAR
                             final UnsafeBeanProperty<Object, Object> beanProperty = (UnsafeBeanProperty<Object, Object>) property.get();
-                            newProperties.set(i, new BeanIntrospectionPropertyWriter(
-                                    existing,
-                                    beanProperty,
-                                    existing.getSerializer(),
-                                    config.getTypeFactory(),
-                                    existing.getViews()
-                                )
-                            );
+                            if (existing instanceof AnyGetterWriter anyGetterWriter) {
```

Comment:
add to the if statement just above this one instead, which does the same check for XmlBeanPropertyWriter